go-backend/service/session_http.go

```
@@ -0,0 +1,240 @@
+		resp, err := http.PostForm("https://oauth2.googleapis.com/token",
+			url.Values{
+				"code":          {auth[0]},
+				"client_id":     {config.ReadEnvString("GOOGLE_KEY")},
+				"client_secret": {config.ReadEnvString("GOOGLE_SECRET")},
+				"scope":         {""},
+				"grant_type":    {"authorization_code"},
+				"redirect_uri":  {config.ReadEnvString("GOOGLE_REDIRECT")},
+			},
+		)
+
+		// TEST URL (FXIME: Delete this from final prod code, leaving it here for future debugging if needed)
+		// https://accounts.google.com/o/oauth2/auth?client_id=749817093280-6h2emqcqsi84murdsdr543kuemvrlr9t.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A33001%2Fauth%2Fgoogle&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&state=state
+
+		if err != nil {
+			log.Error(ae.ErrAuthCodeRequestFail, "Google didn't like the auth code we sent", err)
+			ae.JSONError(rw, http.StatusInternalServerError, err)
+			return
+		}
+
+		var token OAuthToken
+		payload, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Error(ae.ErrReadingResponseBody, "Error reading response body: "+string(payload), err)
+			ae.JSONError(rw, http.StatusInternalServerError, err)
+			return
+		}
+
+		err = json.Unmarshal(payload, &token)
+		if err != nil {
+			log.Error(ae.ErrJSONParseFail, "json.Unmarshal error on payload "+string(payload), err)
+			ae.JSONError(rw, http.StatusInternalServerError, err)
+			return
+		}
+
+		client := &http.Client{}
+		req, _ = http.NewRequest("GET", "https://www.googleapis.com/oauth2/v2/userinfo", nil)
+		req.Header.Set("Authorization", "Bearer "+token.AccessToken)
+		resp, err = client.Do(req)
+		if err != nil {
+			log.Error(ae.ErrHTTPRequestFailed, "Failure executing HTTP request to https://www.googleapis.com/oauth2/v2/userinfo", err)
+			ae.JSONError(rw, http.StatusInternalServerError, err)
+			return
+		}
+
+		user := OAuthUser{}
+		payload, _ = ioutil.ReadAll(resp.Body)
```

again should not ignore the error

---

fa2483e